addon_service/common/hmac.py

```diff
@@ -67,37 +54,145 @@ def make_signed_headers(
     )
 
 
-def _reconstruct_string_to_sign_from_request(request, signed_headers: list[str]) -> str:
-    signed_segments = [request.method, request.path]
-    query_string = request.META.get("QUERY_STRING")
-    if query_string:
-        signed_segments.append(query_string)
-    signed_segments.extend(
-        [str(request.headers[signed_header]) for signed_header in signed_headers]
-    )
-    return "\n".join([segment for segment in signed_segments if segment])
-
-
-def validate_signed_headers(request, hmac_key=None):
-    match = _AUTH_HEADER_REGEX.match(request.headers.get("Authorization", ""))
+@functools.lru_cache
+def validate_signed_headers(
```

Comment:
Nit, because it feels unusual to have a `validate...` function return a result (and be called many times).
```suggestion
def get_hmac_signed_headers(
```

Option b (for better?):
Leave `validate_signed_headers` without a return value but still cache the result. Add a separate `get_signed_headers` function that checks `validate_signed_headers` and returns only the headers that are part of the hmac signature.

Reply:
how about rename to `get_signed_headers` (and keep cache), then add `validate_signed_request` that calls `get_signed_headers` and returns nothing? (this is what i have done, open to thoughts)

Reply:
works for me